### PR TITLE
Speed-up canonical bundle thanks to @HechtiDerLachs

### PR DESCRIPTION
**Pull request by wdecker**

### CODE REVIEW

File: src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl [hunk 1]
```diff
@@ -199,22 +199,40 @@ SubquoModule{MPolyDecRingElem{fpFieldElem, fpMPolyRingElem}}
...
-   Omega = homology(D, 0);
-   Omega, _ = prune_with_map(Omega);  
-   return Omega
+  Pn = ambient_coordinate_ring(X)
+  n = ngens(Pn)-1
+  I = defining_ideal(X)
+  c = codim(I)
+  A, _ = quo(Pn, I)
+  FA = free_resolution(A, algorithm = :fres)
+  C = Oscar.SimpleComplexWrapper(FA.C[0:first(range(FA.C))])
```
Comment (HechtiDerLachs) [inline src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl]:
```suggestion
  Pn = ambient_coordinate_ring(X)
  n = ngens(Pn)-1
  c = codim(X)
  A = homogeneous_coordinate_ring(X)
  C, _ = free_resolution(Oscar.SimpleFreeResolution, A)
```

Comment (fingolfin) [inline src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl:207]:
```suggestion
  C = SimpleComplexWrapper(FA.C[0:first(range(FA.C))])
```

Just to be explicit: `SimpleComplexWrapper` is from `experimental`. Normally code in `src` should not depend on `experimental` code, but apparently this improvement is critical for a book example, plus hopefully `experimental/DoubleAndHyperComplexes` can migrate to `src` in the not-so distant future, so we could make an exception here.

File: src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl [hunk 2]
```diff
@@ -199,22 +199,40 @@ SubquoModule{MPolyDecRingElem{fpFieldElem, fpMPolyRingElem}}
...
+###   FAC = FA.C
+###   r = range(FAC)
+###   C = shift(FAC[first(r):-1:1], -c)
+###   F = free_module(Pn, 1)
+###   OmegaPn = grade(F, [n+1])
+###   D = hom(C, OmegaPn);
+###   Omega = homology(D, 0);
+###   Omega, _ = prune_with_map(Omega);  
+###   return Omega
+###end
```
Comment (fingolfin) [inline src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl]: No need to keep the old code around, we have git for that.
```suggestion
```

File: src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl [hunk 3]
```diff
@@ -199,22 +199,40 @@ SubquoModule{MPolyDecRingElem{fpFieldElem, fpMPolyRingElem}}
...
+  c = codim(I)
+  A, _ = quo(Pn, I)
+  FA = free_resolution(A, algorithm = :fres)
+  C = Oscar.SimpleComplexWrapper(FA.C[0:first(range(FA.C))])
+  C_simp = simplify(C);
+  C_shift = shift(C_simp, c);
+  OmegaPn = graded_free_module(Pn, [n+1])
+  D = hom(C_shift, OmegaPn);
+  D_simp = simplify(D);
+  #return simplify(homology(D, 0)[1])[1]
```
Comment (fingolfin) [inline src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl]: Optional: get rid of the semicolons and the commented out line by applying this patch
```suggestion
  C_simp = simplify(C)
  C_shift = shift(C_simp, c)
  OmegaPn = graded_free_module(Pn, [n+1])
  D = hom(C_shift, OmegaPn)
  D_simp = simplify(D)
```

File: src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl [hunk 4]
```diff
@@ -199,21 +199,20 @@ SubquoModule{MPolyDecRingElem{fpFieldElem, fpMPolyRingElem}}
...
+  FA = free_resolution(A, algorithm = :fres)
+  C = Oscar.SimpleComplexWrapper(FA.C[0:first(range(FA.C))])
+  C_simp = simplify(C)
+  C_shift = shift(C_simp, c)
+  OmegaPn = graded_free_module(Pn, [n+1])
+  D = hom(C_shift, OmegaPn)
+  D_simp = simplify(D)
+  Z, inc = kernel(D_simp, 0)
+  B, inc_B = Oscar.boundary(D_simp, 0)
+  return prune_with_map(SubquoModule(D_simp[0], ambient_representatives_generators(Z), ambient_representatives_generators(B)))[1]
```
Comment (jankoboehm) [inline src/AlgebraicGeometry/Surfaces/AdjunctionProcess/AdjunctionProcess.jl:215]: Eventually we should probably hide these tricks.

**Review by jankoboehm — APPROVED** [review]

**Comment by HechtiDerLachs** [comment]
Just out of curiosity: Does this now finish with `prune_with_map` for `X = rational_d10_pi8()`?

**Comment by wdecker** [comment]
No!

**Pull request merged**